docs/appendices/serialisation.md

```diff
@@ -59,7 +59,7 @@ actor Main
 
 There are several things to keep in mind when using Pony's serialisation system:
 
-* Serialised objects will currently only work when passed between two running instances of the same Pony executable. You cannot pass objects between different Pony programs, nor can you pass them between different versions of the same program.
+* Serialised objects will currently only work when passed between two running instances of the same Pony executable. Passing objects between different Pony programs or between different versions of the same program may lead to a crash or unsafe memory manipulation. You might use the `Serialise.signature` function to determine that sharing two different Pony programs is safe.
```

Comment:
While this is technically true. It is highly unlikely that will ever be true. I think the "less true" version is "more true in practice".

Reply:
Do you mean the "crash" part, or the part about using the signature to know if it is safe?

Reply:
Also, just noticed that there's a missing "data" in the last sentence

Reply:
Technically you might be able to pass objects between different versions of a program but it is highly unlikely. In both those instances, the chance that the type ids are the same is really low. So that "may lead to" is more like "will almost assuredly".

So while your rewrite is technically more correct, I think it is less practically correct. I dont think the rewrite properly conveys the danger and makes it sound easier to accomplish than it is. Current serialization is not intended to be used in the way your rewrite suggests. It was intended in the "only like this" that the original version conveys.

Reply:
Okay, I've reverted to the original version, but added a note about using the `signature` function.